### DISCUSSION TITLE
How do i access vert.x port number?

### DISCUSSION THREAD

**Discussion by Novke** [quarkusio/quarkus#42626] [Q&A] [answered]
Im trying to register microservice to consul discovery service. I have not been able to do it automatically so i do it explicitly using init method with @Observes annotation. Problem arrives when i set port number to be 0 (aka random) because im unable to get the port, as everything i tried to do just returns 0 instead of an actual port number.

Example:

 ```
@ConfigProperty(name = "consul.host")
    String consulAddress;
    @ConfigProperty(name = "consul.port")
    int consulPort;
    Integer applicationPort;
    @ConfigProperty(name = "quarkus.application.name")
    String applicationName;
    ConsulClient consulClient;

public void init(@Observes HttpServer server, Vertx vertx) {
        applicationPort = vertx.createHttpServer().actualPort();
        System.out.println("SERVER PORT IS : " + applicationPort);

        consulClient = ConsulClient.create(vertx, new ConsulClientOptions().setHost(consulAddress).setPort(consulPort));
        consulClient.registerService(
                new ServiceOptions().setPort(applicationPort).setAddress("localhost").setName(applicationName).setId(applicationPort + ""));
    }
```
    
I have also tried HttpConfiguration.port but the result appears to be the same. Any other approach? Anything non-explicit?

!!! It appears that when live reload triggers retart the port is displayed properly. Is there any other event that i can observe that is guaranteed to happen after the port has been set?

**Comment by cescoffier** [comment, marked as answer]
```java
@ConfigProperty(name = "quarkus.http.port", defaultValue="8080") int port;
```

It should be fine, except if you use random ports, you should be able to access it using a system property directly (`quarkus.http.port`).
